lib/news_scraper/trainer/uri_trainer.rb

```diff
@@ -3,60 +3,53 @@ module Trainer
     class UriTrainer
       def initialize(uri)
         @article_scrape_patterns = YAML.load_file(Constants::SCRAPE_PATTERN_FILEPATH)
-        @data_type = Trainer::PresetsSelector.new(uri)
-
-        @uri = uri
-        @raw_html = Extractors::Article.new(uri: uri).extract
-        @transformer = Transformers::Article.new(
-          uri: uri,
-          payload: @raw_html,
-          scrape_patterns: @article_scrape_patterns
-        )
+        uri_parser = URIParser.new(uri)
+        @uri = uri_parser.without_scheme
+        @root_domain = uri_parser.host
+        @raw_html = Extractors::Article.new(uri: @uri).extract
       end
 
       def train
-        if @article_scrape_patterns['domains'].key?(@transformer.root_domain)
-          @transformer.transform
-        else
+        unless @article_scrape_patterns['domains'].key?(@root_domain)
           CLI.put_header
-          CLI.log("There is no scrape pattern defined for #{@transformer.root_domain}"\
+          CLI.log("There is no scrape pattern defined for #{@root_domain}"\
             " in config/article_scrape_patterns.yml")
-          no_scrape_defined(@transformer.root_domain)
+          no_scrape_defined
         end
       end
 
-      def no_scrape_defined(root_domain)
-        if CLI.confirm("Step through presets for #{root_domain}?")
+      def no_scrape_defined
+        if CLI.confirm("Step through presets for #{@root_domain}?")
           CLI.put_footer
-          selected_presets = @data_type.train(@raw_html)
```

Comment:
This line actually caused a lot of trouble because our tests stubbed out `PresetsSelector#train` https://github.com/richardwu/news_scraper/pull/3/files#diff-1f9d04374200e0405dea75c16e01751fL47.

I think we need to refactor our tests and public/private interface so that we get true test coverage for our public interface.